frontend/src/components/HeaderNav.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component, useState, useRef, useEffect } from 'react';
+import { StyleSheet, View, BackHandler } from 'react-native';
+import { Appbar, Searchbar } from 'react-native-paper';
+import { connect, useDispatch } from 'react-redux';
+import StoreSelector from './StoreSelectorButton';
+// import {  } from './SuggestionList';
+import { headerStates, globalStates } from '../constants/States';
+import { State } from '../constants/Interfaces';
+import { actionTypes } from '../constants/ActionTypes';
+
+interface Props {
+    currState: headerStates
+}
+
+function HeaderNav(props: Props) {
+    let [clearQuery, setClearQuery] = useState(() => false);
+    let [query, setQuery] = useState(() => "");
+
+    const dispatch = useDispatch();
+    
+    useEffect(() => {
+        function backButtonPressed() {
+            dispatch({type: globalStates.listSelection});
+            return true;
+        }
+
+        BackHandler.addEventListener('hardwareBackPress', backButtonPressed);
+
+        return () => {
+            BackHandler.removeEventListener('hardwareBackPress', backButtonPressed);
+        }
+    })
+    
+    const backToItemList = () => {
+        dispatch({type: globalStates.listSelection});
+    }
+
+    let result = (
+        <Appbar.Header>
+            <Appbar.Content title="GrocerEZ"/>
+        </Appbar.Header>
+    );
+    switch (props.currState) {
```

Comment:
Has this solution been fast for you on your phone? I haven't actually run this code yet, but in the past, when I tried to change header states using Redux, my phone had like a one-second delay between the button press and the change

Reply:
Currently it has been pretty speedy. However, will see if this changes once all the other components have been implemented